src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -348,7 +349,19 @@ export const useBalanceChartData: UseBalanceChartData = args => {
   const cosmosCaip10 = account ? caip10.toCAIP10({ caip2: cosmosCaip2, account }) : ''
 
   // load staking data to redux state
-  useGetStakingDataQuery({ accountSpecifier: cosmosCaip10 })
+  useEffect(() => {
+    ;(async () => {
+      if (!cosmosCaip10?.length) return
```

Comment:
maybe we should move this check inside the `getStakingData` api also, so if we forget to guard it won't make the network request

Reply:
The reason I didn't add it there is because RTK expects some specific shape of happy/unhappy returns so we can't just early return undefined.
`fromCaip10` already throws currently and we re-throw in https://github.com/shapeshift/web/blob/develop/src/state/slices/stakingDataSlice/stakingDataSlice.ts#L140 so I don't think we could technically return an error shape on no caip10, but we'd be effectively paraphrasing the `fromCaip10` error.
What I think might be useful though is `console.error` logging the error itself, since we currently re-throw without logging it (logging it is what allowed me to debug this in the first place).

Reply:
Error re-thrown in [#ad1d86fe0a670f4e089fff32f66db1fe50d57096](https://github.com/shapeshift/web/pull/1456/commits/ad1d86fe0a670f4e089fff32f66db1fe50d57096)

Reply:
we can early return - see here how to do it

https://github.com/shapeshift/web/blob/013226910fbdddb5bc96a691e3506f6b22eea597/src/state/slices/txHistorySlice/txHistorySlice.ts#L256

Reply:
Improved by early returning the RTK query error shape + `console.error`ing in case of empty/undefined `accountSpecifier` (https://github.com/shapeshift/web/pull/1456/commits/4ba40202073787111e5ccaf28349cb7b289f3564)